subprojects/base-services/src/main/java/org/gradle/internal/resources/DefaultResourceLockCoordinationService.java

```diff
@@ -48,7 +48,7 @@ public boolean withStateLock(Transformer<ResourceLockState.Disposition, Resource
                         case RETRY:
                             resourceLockState.releaseLocks();
                             try {
-                                lock.wait();
+                                lock.wait(5000);
```

Comment:
Is this required to avoid test hanging on CI?

Reply:
It is not. We did that to see if the lock is not notified for some reason or if the problem is something else. We don't need to timeout the lock.

@big-guy WDYT? Should we leave in the timeout?

Reply:
Let's not port this then.